Issue with incremental builds

### Check duplicate issues.

- [x] Checked for duplicates

### Description

Hello,

I compiled ROOT master branch from yesterday (build from source) with:
```
cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo -DPython3_ROOT_DIR=/usr/bin/python3 -Dxrootd=OFF -Dmathmore=ON -Ddavix=OFF -Dbuiltin_tbb=ON  -DCMAKE_INSTALL_PREFIX=../install/ ../root_src/
```

I rebased the source code to today's version and then just ran:
```
cmake --build . --target install -j10
```

However, the build fails with:
```
[ 63%] Generating G__MultiProc.cxx, ../../lib/MultiProc.pcm
In file included from input_line_7:10:
/home/tomas/ROOT/build/include/MPSendRecv.h:18:10: remark: building module 'Net' as '/home/tomas/ROOT/build/lib/Net.pcm' [-Rmodule-build]
#include "TSocket.h"
         ^
Error: Building module 'Net' implicitly. If 'Net' requires a 
dictionary please specify build dependency: 'G__MultiProc.cxx' depends on 'Net'.
Otherwise, specify '-mByproduct Net' to disable this diagnostic.
[ 63%] Generating G__SQLIO.cxx, ../../lib/SQLIO.pcm
[ 63%] Generating G__RootAuth.cxx, ../../lib/RootAuth.pcm
[ 63%] Building CXX object net/rpdutils/CMakeFiles/SrvAuth.dir/src/DaemonUtils.cxx.o
[ 63%] Building CXX object net/rpdutils/CMakeFiles/SrvAuth.dir/src/rpdutils.cxx.o
gmake[2]: *** [core/multiproc/CMakeFiles/G__MultiProc.dir/build.make:94: core/multiproc/G__MultiProc.cxx] Error 1
gmake[1]: *** [CMakeFiles/Makefile2:52802: core/multiproc/CMakeFiles/G__MultiProc.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[ 63%] Built target G__MathCore
```

This seems like some dependancies are not properly rebuilt with an incremental update?

This is my gcc version:
```
gcc (Ubuntu 15.2.0-4ubuntu4) 15.2.0
```

### Reproducer

Repeat the build setup above with changes in the source code that affect ROOT dictionaries 

### ROOT version

commit: 5e6a3597b0ebc060468653aea749fe84239e2449



### Installation method

Build from source

### Operating system

Ubuntu 25.10

### Additional context

_No response_

--- a/cmake/modules/RootMacros.cmake
+++ b/cmake/modules/RootMacros.cmake
@@ -623,22 +623,17 @@ function(ROOT_GENERATE_DICTIONARY dictionary)
   #---Get the library and module dependencies-----------------
   if(ARG_DEPENDENCIES)
     foreach(dep ${ARG_DEPENDENCIES})
-      if(NOT TARGET G__${dep})
-        # This is a library that doesn't come with dictionary/pcm
-        continue()
-      endif()
-
+      # Whether <dep> provides a dictionary/pcm is decided at generation time
+      # via $<TARGET_EXISTS:G__<dep>>, so the '-m' flag and the module-file
+      # dependency below are independent of configuration order and expand to
+      # nothing for a dictionary-less library.
+      set(dep_has_dict "$<TARGET_EXISTS:G__${dep}>")
       set(dependent_pcm ${libprefix}${dep}_rdict.pcm)
       if (runtime_cxxmodules AND NOT dep IN_LIST local_no_cxxmodules)
         set(dependent_pcm ${dep}.pcm)
-        if(TARGET ${dep})
-          get_target_property(_dep_pcm_filename ${dep} ROOT_PCM_FILENAME)
-          if(_dep_pcm_filename)
-            list(APPEND pcm_dependencies ${_dep_pcm_filename})
-          endif()
-        endif()
+        list(APPEND pcm_dependencies "$<${dep_has_dict}:$<TARGET_PROPERTY:${dep},ROOT_PCM_FILENAME>>")
       endif()
-      set(newargs ${newargs} -m  ${dependent_pcm})
+      set(newargs ${newargs} "$<${dep_has_dict}:-m>" "$<${dep_has_dict}:${dependent_pcm}>")
     endforeach()
   endif()
 

